layer/VkLayer_FROG_gamescope_wsi.cpp

```diff
@@ -493,6 +535,18 @@ namespace GamescopeWSILayer {
       if (!gamescopeSurface)
         return pDispatch->GetPhysicalDeviceSurfaceCapabilities2KHR(physicalDevice, pSurfaceInfo, pSurfaceCapabilities);
 
+      // Incomplete writes here, do not return VK_INCOMPLETE.
+      if (auto state = GamescopeSurface::get(pSurfaceInfo->surface)) {
+        if (gamescopeIsForcingFifo() && state->frameLimiterAware()) {
+          static constexpr std::array<VkPresentModeKHR, 1> s_FifoModes = {
+            VK_PRESENT_MODE_FIFO_KHR,
```

Comment:
pSurfaceInfo must have a present mode info pNext. The more "correct" thing to do is to only return that present mode in the output.

Reply:
Surely we'd return 0 present modes for non-FIFO if we are limiting in this case, no?

Reply:
That is out of spec since spec says that the input presentMode must: be returned in the compat list, but it would also be out of spec for application to ask for a present mode that wasn't reported as supported. It just feels more correct to report 1:1 since it mirrors the intention of EXT_swapchain_maint1 spec language.